wire-ios/Wire-iOS/Sources/UserInterface/UserProfile/Devices/ProfileClientViewController.swift

```diff
@@ -453,31 +450,32 @@ final class ProfileClientViewController: UIViewController, SpinnerCapable {
 
     @objc
     private func onDeleteDeviceTapped() {
+        let clientObjectID = self.viewModel.userClient.objectID
         let sync = viewModel.userClient.managedObjectContext!.zm_sync!
-        sync.performGroupedBlockAndWait { [weak self] in
-            guard let weakSelf = self else { return }
-
-            let client = try! sync.existingObject(with: weakSelf.viewModel.userClient.objectID) as! UserClient
-            client.deleteClientAndEndSession()
-            sync.saveOrRollback()
+        Task { [self] in
+            let client = await sync.perform { try! sync.existingObject(with: clientObjectID) as! UserClient }
```

Comment:
can we use try? instead

Reply:
also add some handling I guess if we have no client we should reload the list no?

Reply:
I this is a debug function only visible to developers, I'm sure we can improve this but I don't this PR is the place for it.